src/main.js

```diff
@@ -1,144 +1,175 @@
+    // this.subscriptions = new CompositeDisposable()
+    this.subscriptions = makeCompositeDisposable()
+
+    // Put dependency loading on Atom's todo-list
+    this.subscriptions.add(makeIdleRunner()(dependencies))
+
+    const {
+      atomConfig,
+      getMigrations,
+      subscribe: configSubscribe
+    } = require('./atom-config')
+
+    // Subscribe to Atom configuration settings
     this.subscriptions.add(...(configSubscribe()))
 
+    // Subscribe to Save event so we can run fixOnSave
     this.subscriptions.add(atom.workspace.observeTextEditors((editor) => {
       editor.onDidSave(async () => {
+        const { hasValidScope } = require('./validate/editor')
+
         const { fixOnSave, scopes } = atomConfig
         if (hasValidScope(editor, scopes) && fixOnSave) {
           await this.fixJob(true)
         }
       })
     }))
 
+    // Subscribe to Debug command
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
       'linter-eslint:debug': async () => {
-        loadDeps()
-        const debugString = await debug.report()
-        const notificationOptions = { detail: debugString, dismissable: true }
+        const { report } = require('./debug')
+
+        const debugReport = await report()
+        const notificationOptions = { detail: debugReport, dismissable: true }
         atom.notifications.addInfo('linter-eslint debugging information', notificationOptions)
       }
     }))
 
+    // Subscribe to Fix File command
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
       'linter-eslint:fix-file': async () => {
         await this.fixJob()
       }
     }))
 
+    // Add context menu entry for Fix File
     this.subscriptions.add(atom.contextMenu.add({
       'atom-text-editor:not(.mini), .overlayer': [{
         label: 'ESLint Fix',
         command: 'linter-eslint:fix-file',
         shouldDisplay: (evt) => {
           const activeEditor = atom.workspace.getActiveTextEditor()
-          if (!activeEditor) {
-            return false
-          }
-          // Black magic!
+          if (!activeEditor) return false
+
+          // Some scary voodoo black magic! Atom v1.19.0+
           // Compares the private component property of the active TextEditor
-          //   against the components of the elements
+          //   against the components property of the TextEditor DOM elements
           const evtIsActiveEditor = evt.path.some(elem =>
-            // Atom v1.19.0+
             (elem.component && activeEditor.component
               && elem.component === activeEditor.component))
+
           // Only show if it was the active editor and it is a valid scope
           const { scopes } = atomConfig
+          const { hasValidScope } = require('./validate/editor')
+
           return evtIsActiveEditor && hasValidScope(activeEditor, scopes)
         }
       }]
     }))
 
-    scheduleIdleTasks()
+    // Load miscellaneous idle tasks and potential settings migrations
```

Comment:
This is actually _performing_ the migrations, not loading them.

Reply:
Well this line of code is definitely _not_ performing any migrations. It's simply passing a list of to the idleCallback runner function. I think in my head it was "loading the list of migrations into the runner". Happy to hear how you might word it.